cime_config/machines/cmake_macros/crayclang_crusher.cmake

```diff
@@ -1,9 +1,3 @@
-if (compile_threaded)
```

Comment:
why is this block removed for cpu build, but below, for gpu build, it is added?

Reply:
`-fopenmp` gets added in the higher-level `crayclang.cmake`.
Ideally, `crayclang_crusher.cmake` should contain Crusher-specific flags (and avoid duplicates).

I didn't see a prior `crayclanggpu.cmake` (only Crusher had `crayclanggpu` at the time) and just added a single `crayclanggpu_crusher.cmake` that had both generic and machine-specific flags.
I can rebase onto the latest master and update the branch.